rust/template/src/api/c_api.rs

```diff
@@ -35,45 +35,88 @@ use std::os::unix::io::{FromRawFd, IntoRawFd, RawFd};
 use std::os::windows::io::{FromRawHandle, IntoRawHandle, RawHandle};
 
 #[no_mangle]
-pub unsafe extern "C" fn ddlog_get_table_id(tname: *const raw::c_char) -> libc::size_t {
-    if tname.is_null() {
+pub unsafe extern "C" fn ddlog_get_table_id(
+    prog: *const HDDlog,
+    tname: *const raw::c_char,
+) -> libc::size_t {
+    if prog.is_null() || tname.is_null() {
         return libc::size_t::max_value();
     }
 
+    let prog = Arc::from_raw(prog);
```

Comment:
`ManuallyDrop::new(Arc::from_raw(..))` is less error prone and more clear than calling from_raw and later into_raw